src/main/java/net/datafaker/providers/base/Country.java

```diff
@@ -27,12 +34,22 @@ public String capital() {
         return resolve("country.capital");
     }
 
+    /**
+     * @see Currency#getDisplayName()
+     * @return a random detailed ISO 4217 currency display name
+     */
     public String currency() {
-        return resolve("country.currency");
+        int randomIndex = faker.random().nextInt(availableCurrencies.size());
+        return availableCurrencies.get(randomIndex).getDisplayName();
```

Comment:
Is there a reason for the intermediate store when it could just be inlined?

Reply:
That would be quite a long line don't you think?

Reply:
🤷‍♂️ spotless should take care of it